apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -158,76 +140,51 @@ const LineDiagram = ({
+        {lineDiagram.map(
+          ({
+            stop_data: stopData,
+            route_stop: routeStop,
+            alerts: stopAlerts
+          }: LineDiagramStop) => (
+            <div key={routeStop.id} className="m-schedule-diagram__stop">
               <div
                 style={{ color: `#${route.color}` }}
                 className="m-schedule-diagram__lines"
               >
-                <div className="m-schedule-diagram__line">
-                  {!isOnSecondaryBranch(lineDiagramStop) && (
-                    <svg>
-                      <circle
-                        r="4"
-                        cx="5"
-                        cy={isBranchStart(index) ? "2" : "26"}
-                      />
-                    </svg>
-                  )}
-                </div>
-
-                {isOnSecondaryBranch(lineDiagramStop) && (
-                  <div className="m-schedule-diagram__line m-schedule-diagram__line--branch">
-                    <svg>
-                      {isMergeStop(lineDiagramStop) && (
-                        <path
-                          d={`
-                            M-10,${isBranchStart(index) ? "-30" : "100"}
-                            h 15 
-                            v ${isBranchStart(index) ? "36" : "-76"}
-                          `}
-                        />
-                      )}
-                      <circle
-                        r="4"
-                        cx="5"
-                        cy={isBranchStart(index) ? "2" : "26"}
-                      />
-                    </svg>
+                {stopData.map((sd, sdIndex) => (
+                  <div
+                    key={`${routeStop.id}-${sd.type}-${sd.branch}`}
+                    className={`m-schedule-diagram__line m-schedule-diagram__line--${
+                      sd.type
+                    }`}
+                  >
+                    {sdIndex + 1 === stopData.length && (
+                      <svg
+                        viewBox="0 9 10 10"
+                        height="10"
+                        className="m-schedule-diagram__line-stop"
+                      >
+                        {sd.type === "merge" && (
+                          <path
+                            className="m-schedule-diagram__line-bend"
+                            d="M-15,-30 h 20 v 60"
+                          />
+                        )}
+                        <circle r="4" cx="50%" />
+                      </svg>
+                    )}
                   </div>
-                )}
+                ))}
               </div>
               <div className="m-schedule-line-diagram__content">
                 <div className="m-schedule-line-diagram__card">
                   <div className="m-schedule-line-diagram__card-left">
                     <div className="m-schedule-line-diagram__stop-name">
-                      {branchName(lineDiagramStop)}
+                      {StopBranchLabel(routeStop)}
```

Comment:
Since this is defined like a component, should we invoke it as `<StopBranchLabel stop=routeStop>`?

Reply:
Just realized I forgot to respond to this -- my thought on this is that this function generates boring JSX that doesn't need to be React-ified so to speak -- there doesn't seem to be anything to gain by defining a React component with props and whatnot for this :)